Added option to use self-signed certificates

anrudolph opened the pull request:
## Description:

I defined a new option for configuration.yaml, 'verify_ssl', which is set to 'True' by default for obvious security reasons. However, in order to work with self-signed certificates, it is now possible to disable the certificate validation.(eg, I use a nextcloud instance with self-signed certificates)
Credit for code in line 57 goes to user 'gen2' on the homeassistant community, who suggested this solution. 
https://community.home-assistant.io/t/caldav-configuration/38198/25
I only took it a step further and made it a config option.

## Example entry for `configuration.yaml` (if applicable):
```yaml
calendar:
  - platform: caldav
    username: <myname>
    password: <mysupersecretpassword>
    url: <nextcloudurl>
    verify_ssl: False
    
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [x] There is no commented out code in this PR.

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If the code communicates with devices, web services, or third-party tools:
  - [ ] [_The manifest file_][manifest-docs] has all fields filled out correctly ([example][ex-manifest]).
  - [ ] New dependencies have been added to `requirements` in the manifest ([example][ex-requir]).
  - [ ] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [ ] New or updated dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [ ] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Tests have been added to verify that the new code works.

[ex-manifest]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/mobile_app/manifest.json
[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/mobile_app/manifest.json#L5
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard/__init__.py#L23
[manifest-docs]: https://developers.home-assistant.io/docs/en/development_checklist.html#_the-manifest-file_

anrudolph committed:
Added option to use self-signed certificates

I defined a new option for configuration.yaml, 'verify_ssl', which is set to 'True' by default for obvious security reasons. However, in order to work with self-signed certificates, it is now possible to disable the certificate validation.(eg, I use a nextcloud instance with self-signed certificates)
Credit for code in line 57 goes to user 'gen2' on the homeassistant community, who suggested this solution. 
https://community.home-assistant.io/t/caldav-configuration/38198/25
I only took it a step further and made it a config option.

homeassistant commented:
Hi @anrudolph, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#23139).

Once you do that we will be able to review and accept this pull request.

Thanks!

anrudolph committed:
Update calendar.py

anrudolph committed:
Added the import of CONF_VERIFY_SSL

I hope this passes the test now...

awarecan commented on homeassistant/components/caldav/calendar.py:
```diff
@@ -50,7 +51,10 @@ def setup_platform(hass, config, add_entities, disc_info=None):
     password = config.get(CONF_PASSWORD)
 
-    client = caldav.DAVClient(url, None, username, password)
+    if config.get(CONF_VERIFY_SSL):
```
```suggestion
    client = caldav.DAVClient(url, None, username, password,
                              ssl_verify_cert=config[CONF_VERIFY_SSL])
```

awarecan reviewed (commented):
Could be simplify

awarecan committed:
Update homeassistant/components/caldav/calendar.py

Cool! Didn't know about that possibility, my coding experience is literally two weeks. Thanks for sharing!

Co-Authored-By: anrudolph <49680492+anrudolph@users.noreply.github.com>

anrudolph committed:
Removed some lines of code

I think, in order for the last commit to work, these lines have to be removed. Correct?

anrudolph committed:
Trying to get this passing the checks

Trying around to get the simplified code to work
